juju/juju
# [JUJU-1985] Support juju 3.0 controllers and centos series

Juju 3 controllers correctly model centos 7 as a base with os name "centos" and channel "7".
Juju 2.9 still uses "centos7".

This PR adapts the 2.9 client facades to correctly send/interpret the centos channel depending on what version controller is being used. The relevant facades have had a version increment so that the client can distinguish which ones are used. The server side facade methods are unchanged but we still need to registered the updated versions.

The instance config used by cloudinit is updated to use a base, not series (backported from 3).

The addition of base to juju status output is reverted so we retain what was previously displayed in 2.9.

Also, The UpdateApplicationSeries method is renamed to UpdateApplicationBase in 3.0, so account for that also.

## Checklist


## QA steps

Test 4 combinations.
This PR client -> 2.9.35 controller
This PR client -> this PR controller
This PR client -> 3.0-rc2 controller
2.9.35 client -> this PR controller

The 2 key operations are adding a centos7 machine and deploying a centos7 charm.
For new controllers, the centos agents will not be found but the cli deploy / add-machine will work and inspecting the db will show the expected series and/or platform recorded. ie "centos7" for older controllers and "centos/7" for newer.

I used `aws/ap-southeast-2` region and
`juju metadata add-image --series centos7 ami-0cf5f53cea16d8cbf` to teach juju the centos7 image id.

Then the operations are
```
juju add-machine --series=centos7
juju deploy slurmd --series=centos7
```

Also juju create-backup
